### PR TITLE
Collision drawables shared buffers

### PR DESCRIPTION
**Pull request by stefankarschti**
Switch collision boxes and circles to shared vertex buffers and fix a few render tests failures. 

Ref #1672 
Related to #1855 

Before:
Before

After:
After

### CODE REVIEW

**Review by TimSylvester — APPROVED** [review]

**Review by alexcristici — APPROVED** [review]

**Review by sjg-wdw — APPROVED** [review]

File: src/mbgl/mtl/vertex_attribute.cpp [hunk 1]
```diff
@@ -4,78 +4,303 @@
...
+                break;
+            case gfx::AttributeDataType::UInt:
+                get<uint32_t>(value);
+                break;
+            case gfx::AttributeDataType::Float:
+                get<float>(value);
+                break;
+            default:
+                assert(false);
+                buffer += stride;
```
Comment (louwers) [inline src/mbgl/mtl/vertex_attribute.cpp:49]: Could you explain these?

File: src/mbgl/mtl/vertex_attribute.cpp [hunk 2]
```diff
@@ -4,78 +4,303 @@
...
+        buffer += stride;
+    }
+
+private:
+    gfx::AttributeDataType dataType;
+    std::uint8_t* buffer{nullptr};
+    std::size_t stride{0};
+
+    template <typename To, typename From, typename = std::enable_if_t<std::is_assignable_v<To&, From>>>
+    void get(const From value) {
```
Comment (louwers) [inline src/mbgl/mtl/vertex_attribute.cpp:283]: `get` is not a good name for what this does.

Comment (louwers) [reply src/mbgl/mtl/vertex_attribute.cpp:283]: `set` seems more applicable?

File: src/mbgl/mtl/vertex_attribute.cpp [hunk 3]
```diff
@@ -4,78 +4,303 @@
...
+        assert(stride == sizeof(To));
+        *reinterpret_cast<To*>(buffer) = static_cast<To>(value);
+        buffer += stride;
+    }
+
+    template <typename To,
+              typename From,
+              std::size_t Size,
+              typename = std::enable_if_t<std::is_assignable_v<To&, From>>>
+    void get(const std::array<From, Size>& value) {
```
Comment (louwers) [inline src/mbgl/mtl/vertex_attribute.cpp:293]: This one does not seem to be used. YANGI?

File: src/mbgl/mtl/vertex_attribute.cpp [hunk 4]
```diff
@@ -4,78 +4,303 @@
...
+
+private:
+    gfx::AttributeDataType dataType;
+    std::uint8_t* buffer{nullptr};
+    std::size_t stride{0};
+
+    template <typename To, typename From, typename = std::enable_if_t<std::is_assignable_v<To&, From>>>
+    void get(const From value) {
+        assert(stride == sizeof(To));
+        *reinterpret_cast<To*>(buffer) = static_cast<To>(value);
```
Comment (louwers) [inline src/mbgl/mtl/vertex_attribute.cpp:285]: I know this is just an internal helper class, but this seems asking for trouble. It takes one off by one error or one call to many to operator() to introduce an out of bounds write.

Adding this functionality to `mtl::VertexAttribute` may be a more memory safe alternative to consider.

File: src/mbgl/mtl/vertex_attribute.cpp (hunk 2 above)
Comment (stefankarschti) [reply src/mbgl/mtl/vertex_attribute.cpp:283]: sure, I'll make a few renames to make it clearer.

**Comment by stefankarschti** [comment]
I think we don't need static vertex attribute buffer building anymore. I removed it to save on size.

**Pull request merged**